server/recceiver/cfstore.py

```diff
@@ -48,125 +55,154 @@ def commit(self, TR):
         else:
-            _log.error('failed to initialize one or more of the following properties'+
-                       'hostname: %s iocname: %s owner: %s',hostName,iocName,owner)
-            
-def updateChannelFinder(client, pvNames, hostName, iocName, time, owner):
-    '''
-    pvNames = list of pvNames 
-    ([] permitted will effectively remove the hostname, iocname from all channels)
-    hostName = pv hostName (None not permitted)
-    iocName = pv iocName (None not permitted)
-    owner = the owner of the channels and properties being added, this can be different from the user
-    e.g. user = abc might create a channel with owner = group-abc
-    time = the time at which these channels are being created/modified
-    '''
-    if hostName == None or iocName == None:
+            _log.error('failed to initialize one or more of the following properties' +
+                       'hostname: %s iocname: %s owner: %s', hostName, iocName, owner)
+
+def __updateCF__(client, new, delrec, channels_dict, iocs, hostName, iocName, time, owner):
+    if hostName is None or iocName is None:
         raise Exception('missing hostName or iocName')
     channels = []
     checkPropertiesExist(client, owner)
-    previousChannelsList = client.findByArgs([('hostName', hostName), ('iocName', iocName)])
-    if previousChannelsList != None:
-        for ch in previousChannelsList:
-            if pvNames != None and ch[u'name'] in pvNames:
-                channels.append(updateChannel(ch,\
-                                              owner=owner, \
-                                              hostName=hostName, \
-                                              iocName=iocName, \
-                                              pvStatus='Active', \
-                                              time=time))
-                pvNames.remove(ch[u'name'])
-            elif pvNames == None or ch[u'name'] not in pvNames:
-                '''Orphan the channel : mark as inactive, keep the old hostName and iocName'''
-                oldHostName = hostName
-                oldIocName = iocName
-                oldTime = time
-                for prop in ch[u'properties']:
-                    if prop[u'name'] == u'hostName':
-                        oldHostName = prop[u'value']
-                    if prop[u'name'] == u'iocName':
-                        oldIocName = prop[u'value']
-                    if prop[u'name'] == u'time':
-                        oldTime = prop[u'value']
-                channels.append(updateChannel(ch, \
-                                owner=owner, \
-                                hostName=oldHostName, \
-                                iocName=oldIocName, \
-                                pvStatus='InActive', \
-                                time=oldTime))
+    old = client.findByArgs([('hostName', hostName), ('iocName', iocName)])
+    if old is not None:
+        for ch in old:
```

Comment:
FYI, since there is no 'else' here and ```old``` is not use again, you can abbreviate these three lines as:

```py
    for ch in client.findByArgs([('hostName', hostName), ('iocName', iocName)]) or []
```